packages/SketchMorph2-Core.package/M2Backend.class/instance/clickAt..st

```diff
@@ -1,5 +1,18 @@
 drawing
 clickAt: aPoint
 	self isInFillMode 
-		ifTrue: [self fillShapeAt: aPoint]
-		ifFalse: [self startDrawingAt: aPoint]
\ No newline at end of file
+		ifTrue: [
+			self fillShapeAt: aPoint.
+			^ self].
+	(self isInPenMode or: [self isInGeometricMode])
+		ifTrue: [
+			self startDrawingAt: aPoint.
+			^ self].
+	self isInTextMode
+		ifTrue: [
+			self startEditingTextAt: aPoint.
+			^ self].
+	self isInEditingTextMode
+		ifTrue: [
+			self printText.
```

Comment:
Personally, I'd prefer that clicking anywhere on the canvas automatically adds a new text field when currently editing one.

```suggestion
			self
				printText;
				startEditingAt: aPoint.
```

Reply:
Works as in MS Paint, will remain as-is.